ipatests/test_integration/test_adtrust_install.py

```diff
@@ -303,3 +304,570 @@ def test_ipa_user_s4u2self_pac(self):
+                netbios_name,
+                "-U",
+            ],
+            raiseonerr=False,
+        )
+        assert msg in result.stdout_text
+        assert result.returncode == 0
+
+    def test_adtrust_install_with_non_ipa_user(self):
+        """
+        Test that ipa-adtrust-install command works
+        when kinit is done as alias i.e root which is
+        not an ipa user.
+        """
+        msg = (
+            'Unrecognized error during check of admin rights: '
+            'root: user not found'
+        )
+        user = 'root'
+        self.master.run_command(
+            ["kinit", "-E", user],
+            stdin_text=self.master.config.admin_password,
+            raiseonerr=False
+        )
+        result = self.master.run_command(
+            ["ipa-adtrust-install", "-A", user,
+             "-a", self.master.config.admin_password,
+             "-U"], raiseonerr=False
+        )
+        assert result.returncode != 0
+        assert msg in result.stderr_text
+
+    def test_adtrust_install_as_regular_ipa_user(self):
+        """
+        Test to check ipa-aadtrust-install as a regular ipa user
+        without admin privileges
+        """
+        user = "ipauser1"
+        passwd = "Secret123"
+        try:
+            tasks.kinit_admin(self.master)
+            tasks.create_active_user(
+                self.master,
+                user,
+                password=passwd,
+                first=user,
+                last=user,
+            )
+            self.master.run_command(["kdestroy", "-A"])
+            tasks.kinit_as_user(self.master, user, passwd)
+            self.master.run_command(["klist", "-l"])
+            result = self.master.run_command(
+                ["ipa-adtrust-install", "-A", user,
+                 "-a", passwd, "-U"], raiseonerr=False
+            )
+            msg = "Must have administrative privileges to " \
+                  "setup AD trusts on server\n"
+            assert msg in result.stderr_text
+            assert result.returncode != 0
+        finally:
+            tasks.kinit_admin(self.master)
+            self.master.run_command(["kdestroy", "-A"])
+
+    def test_adtrust_install_as_admins_group_user(self):
+        """
+        Test to check that ipa-adtrust-install is successfull
+        when a regular ipa user is part of the admins group
+        """
+        user = "testuser1"
+        org_pwd = "Secret123"
+        new_pwd = "Secret.123"
+        tasks.kinit_admin(self.master)
+        pwd_stdin_text = "{pwd}\n{pwd}".format(pwd=org_pwd)
+        self.master.run_command(['ipa', 'user-add', user,
+                                 '--first', user,
+                                 '--last', user,
+                                 '--password'],
+                                stdin_text=pwd_stdin_text)
+        self.master.run_command(
+            ["ipa", "group-add-member", "admins", "--users={}".format(user)]
+        )
+        tasks.kinit_user(
```

Comment:
kinit_user is not needed if you provide -A user -a password to ipa-adtrust-install